File: quotation-service/src/routes/status.ts
import { Router } from 'express';
import { getLlmSettings } from '../db.js';

export const statusRouter = Router();

/** Public, no-secret status — lets the main app show "Moteur LLM: provider · model" without auth. */
statusRouter.get('/', (_req, res) => {
  const settings = getLlmSettings();
  return res.json({
    success: true,
    provider: settings.provider,
    modelName: settings.modelName,
  });
});

File: inference-service/app/routers/status.py
from fastapi import APIRouter

from .. import db

router = APIRouter(prefix="/api", tags=["status"])


@router.get("/health")
def health():
    return {"ok": True, "service": "inference-service"}


@router.get("/status")
def public_status():
    """Public, no-secret status — lets the main app show engine badges without auth."""
    llm = db.get_llm_settings()
    embedding = db.get_embedding_settings()
    return {"success": True, "chatModel": llm["model"], "embeddingModel": embedding["model"]}
